Kernel/Syscalls/mmap.cpp

```diff
@@ -587,6 +587,10 @@ ErrorOr<FlatPtr> Process::sys$msync(Userspace<void*> address, size_t size, int f
     if (address.ptr() % PAGE_SIZE != 0)
         return EINVAL;
 
+    if (Memory::page_round_up_would_wrap(size)) {
+        return EINVAL;
```

Comment:
Maybe we should tell them the error is overflow specifically? 
```suggestion
        return EOVERFLOW;
```

Reply:
Hmm, I like the idea of being specific, but this would make `msync` inconsistent with `mmap`, which returns `EINVAL` at the moment in this situation (unless we change that too). For what it's worth, using this test snippet:

```cpp
#include <stdio.h>
#include <unistd.h>
#include <sys/mman.h>
#include <sys/errno.h>

int main(void) {
    if (msync(0x00000000, 0xffffffff, 0x00000001) == -1) {
        perror("msync failed");
    }
    return 0;
}
```

I get `ENOMEM` on MacOS (I guess either `EINVAL` or `ENOMEM` would make sense according to the [manpage](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/msync.2.html)). So that leaves us with three options: `EINVAL`, `ENOMEM` or `EOVERFLOW` :)

I think I'd lean towards just `EINVAL`, but happy to apply your suggestion too. What do you think?

Reply:
Ok, not important enough to debate 😁